lips/lip-20.md

```diff
@@ -0,0 +1,426 @@
+---
+lip: 20
+title: Libra On-Chain Currency Management
+authors: Tim Zakian, Sam Blackshear
+status: Draft
+type: Informational
+created: 08/28/2020
+---
+
+---
+# Summary
+---
+
+This LIP describes the conceptual model and implementation of currencies on the Libra
+blockchain. Currencies in Libra are represented statically at the type level, each
+currency with a unique Move type. A type may treated as a currency in Libra
+if and only if it has previously been registered as a currency on-chain. Once a currency
+is registered it remains registered forever.
+
+Every currency that is registered on-chain has metadata
```

Comment:
> There should be commas both before and after "i.e." and "e.g." -- you've only got commas after those. It helps if you think of spelling them out as "that is" or "for example".

Good call. I'll update :) 

> I can see why you were tempted to use a semicolon between "a SCS...fiat currency" and "or a synthetic currency..." because there are already lots of commas, but I don't think a semicolon works there. It should be a comma.

Yea...I might look in to rephrasing that. Not having a clean separation there seems not very nice.

> You should have something to separate "metadata associated with it at the time of registration" from "such as the currency-code....". A comma would work, or else you could make that a separate sentence: e.g., "That metadata includes..."

👍 

> Should "currency-code" be hyphenated? I think not.

👍 

> How about "on-chain"? In this case, it depends on the usage (and I know that this is not done consistently, but hey, I'm already writing all this about punctuation!) In a phrase like "currencies registered on chain", I don't believe a hyphen should be used. The "on chain" is a prepositional phrase described where the currencies are registered. If you refer to something like "on-chain resources", the "on-chain" is used as an adjective describing the resources. (There's surely a more precise term for this, but I forget what it is.)

Ahh! yes good point. The rule of thumb I use was given to me by a PL person: use "-" for words that need to be bound to the left... So not nearly as precise as yours 😉 

> ...and then there's the Oxford comma question (before "and other information"). I tend to use those myself and since you're an Oxford grad, I would have expected one here 

They actually just called. They're planning on revoking my degree because of this.... In all seriousness thanks for pointing it out since I am a very big fan (and user!) of the Oxford comma. I'll fix :)